js/src/framework2.ts

```diff
@@ -82,6 +84,90 @@ export class ToolBuilder {
   }
 }
 
+export class ScorerBuilder {
+  private taskCounter = 0;
+  constructor(private readonly project: Project) {}
+
+  public create<
+    Output,
+    Extra,
+    Params extends ScorerArgs<Output, Extra>,
+    Returns,
+    Fn extends GenericFunction<Params, Returns>,
+  >(opts: ScorerOpts<Output, Extra, Params, Returns, Fn>) {
+    this.taskCounter++;
+
+    let resolvedName = opts.name;
+    if (!resolvedName && "handler" in opts) {
+      resolvedName = opts.handler.name;
+    }
+    if (!resolvedName || resolvedName.trim().length === 0) {
+      resolvedName = `Scorer ${path.basename(__filename)} ${this.taskCounter}`;
+    }
+    const slug =
+      opts.slug ?? slugifyLib(resolvedName, { lower: true, strict: true });
+
+    if ("handler" in opts) {
+      const scorer: CodeFunction<Params, Returns, Fn> = new CodeFunction(
+        this.project,
+        {
+          ...opts,
+          name: resolvedName,
+          slug,
+          type: "scorer",
+        },
+      );
+      if (globalThis._lazy_load) {
+        globalThis._evals.functions.push(
+          scorer as CodeFunction<
+            unknown,
+            unknown,
+            GenericFunction<unknown, unknown>
+          >,
+        );
+      }
+    } else {
+      const promptBlock: PromptBlockData =
+        "messages" in opts
+          ? {
+              type: "chat",
+              messages: opts.messages,
+            }
+          : {
+              type: "completion",
+              content: opts.prompt,
+            };
+      const promptData: PromptData = {
+        prompt: promptBlock,
+        options: {
+          model: opts.model,
+          params: opts.params,
+        },
+        parser: {
+          type: "llm_classifier",
+          use_cot: opts.useCot,
+          choice_scores: opts.choiceScores,
+        },
+      };
+      // TODO(sachin): Support tools for LLM scorers.
```

Comment:
LLM scorers don't support tools since the choice scores thing turns into tools. So we shouldn't let ppl write them.

Reply:
Removed it. Also restricted the input type so that the compiler checks that tools are not provided.